src/ert/scheduler/scheduler.py

```diff
@@ -209,79 +187,21 @@ def count_states(self) -> Dict[JobState, int]:
-        headers = Headers()
-        if self._ee_token:
-            headers["token"] = self._ee_token
-        event = None
-        async for conn in connect(
-            self._ee_uri.replace("dispatch", "client"),
-            ssl=tls,
-            extra_headers=headers,
-            max_size=2**26,
-            max_queue=500,
-            open_timeout=5,
-            ping_timeout=60,
-            ping_interval=60,
-            close_timeout=60,
-        ):
-            try:
-                self._consumer_started.set()
-                async for message in conn:
-                    try:
-                        event = from_json(
-                            str(message), data_unmarshaller=evaluator_unmarshaller
-                        )
-                        if event["type"] == EVTYPE_FORWARD_MODEL_CHECKSUM:
-                            self.checksum.update(event.data)
-                    except DataUnmarshallerError:
-                        logger.error(
-                            "Scheduler checksum consumer received unknown message"
-                        )
-            except (ConnectionRefusedError, ConnectionClosed, ClientError) as exc:
-                self._consumer_started.clear()
-                logger.debug(
-                    f"Scheduler connection to EnsembleEvaluator went down: {exc}"
-                )
+        while True:
+            event = await self._manifest_queue.get()
+            if event["type"] == EVTYPE_FORWARD_MODEL_CHECKSUM:
+                self.checksum.update(event.data)
+            self._manifest_queue.task_done()
 
     async def _publisher(self) -> None:
-        if not self._ee_uri:
+        if not self._ee_queue:
             return
-        tls: Optional[ssl.SSLContext] = None
-        if self._ee_cert:
-            tls = ssl.SSLContext(ssl.PROTOCOL_TLS_CLIENT)
-            tls.load_verify_locations(cadata=self._ee_cert)
-        headers = Headers()
-        if self._ee_token:
-            headers["token"] = self._ee_token
-        event = None
-        async for conn in connect(
-            self._ee_uri,
-            ssl=tls,
-            extra_headers=headers,
-            open_timeout=60,
-            ping_timeout=60,
-            ping_interval=60,
-            close_timeout=60,
-        ):
-            try:
-                while True:
-                    if event is None:
-                        event = await self._events.get()
-                    if event == CLOSE_PUBLISHER_SENTINEL:
-                        self._publisher_done.set()
-                    else:
-                        await conn.send(event)
-                    event = None
-                    self._events.task_done()
-            except ConnectionClosed:
-                logger.debug("Connection to EnsembleEvalutor went down, reconnecting.")
-                continue
+        while True:
+            event = await self._events.get()
+            await self._ee_queue.put(event)
```

Comment:
Here we are only moving the events from one queue to the next one. Can we pass `self._ee_queue` to the job/driver instead of `self._events`? Then we could get rid of the `_publisher` task completely

Reply:
Currently it is required, since `_ee_queue` is a consumer a does not need to be set.